### PR TITLE
Features/voice receive

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

File: discord/sinks/core.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+            self.header, self.data
+        )
+        self.decoded_data = None
+
+        self.user_id = None
+
+
+class AudioData:
+    """Handles data that's been completely decrypted and decoded and is ready to be saved to file.
+    .. versionadded:: 2.1
```
Comment (VincentRPS) [inline discord/sinks/core.py:116]:
```suggestion
    
    .. versionadded:: 2.1

```

File: discord/sinks/core.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+        the officially provided sink classes
+        like :class:`~discord.sinks.WaveSink`
+    
+    just replace the following like so: ::
+        vc.start_recording(
+            MySubClassedSink(),
+            finished_callback,
+            ctx.channel,
+        )
+    .. versionadded:: 2.1
```
Comment (VincentRPS) [inline discord/sinks/core.py:169]:
```suggestion
    
    .. versionadded:: 2.1
```

**Comment by VincentRPS** [comment]
closes #52

**Comment by VincentRPS** [comment]
@Lulalaby can you possibly fix the merge conflicts?

**Comment by Lulalaby** [comment]
And can you fix your discord account @VincentRPS

**Comment by VincentRPS** [comment]
> And can you fix your discord account @VincentRPS

I wish I could

**Review by BobDotCom — APPROVED** [review]

File: discord/sinks/core.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-2021 Rapptz
```
Comment (Dorukyum) [inline discord/sinks/core.py:4]: Why is this here? Has this file been moved from somewhere else?

Comment (VincentRPS) [reply discord/sinks/core.py:4]: wdym by "moved"?

Comment (Dorukyum) [reply discord/sinks/core.py:4]: Is it taken/copied from somewhere else?

Comment (VincentRPS) [reply discord/sinks/core.py:4]: it is a enhanced version of the discord.py pull as said from the original PR https://github.com/Pycord-Development/pycord/pull/532

Comment (Dorukyum) [reply discord/sinks/core.py:4]: I see

**Review by Dorukyum — APPROVED** [review]
